### ISSUE TITLE
Erroneous error about language version overrides in parts

### ISSUE THREAD

**Issue by Klikini** [dart-lang/sdk#42661]
When compiling a web application, I get the following output (repeated for every `part` statement)

```
web/main.dart:93:6:
Error: The language version override has to be the same in the library and its part(s).
part 'package:couclient/src/display/windows/go_window.dart';
     ^
```

By Googling that error, I found out that it happens when a file contains `// @dart <version>`. My project does not contain any version overrides, and the only build override in the dependencies is in `bazel_worker` at the top of `lib/src/worker_protocol.pb.dart`:

```
///
//  Generated code. Do not modify.
//  source: worker_protocol.proto
//
// @dart = 2.3
// ignore_for_file: camel_case_types,non_constant_identifier_names,library_prefixes,unused_import,unused_shown_name,return_of_invalid_type
// ignore_for_file: annotate_overrides
```

I installed version 2.3 of the SDK, but webdev does not build under that version, so I had to upgrade back to the latest version.

Is there reason for this version override? Is it possible to remove it or avoid it? I'm not sure how anyone is supposed to build a project using webdev with this issue, unless I'm missing something. Any assistance is appreciated, and I'm happy to provide any more information or debugging if someone can point me in the right direction :)

# Version information

```
$ dart --version
Dart VM version: 2.8.4 (stable) (Wed Jun 3 12:26:04 2020 +0200) on "windows_x64"
```

```
$ webdev --version
2.5.9
```

## Relevant `pubspec.yaml`

```yaml
dev_dependencies:
  build_runner: ^1.6.2
  build_web_compilers: ^2.0.0
  json_serializable: ^3.0.0
```

## Relevant `pubspec.lock`

```yaml
packages:
  bazel_worker:
    dependency: transitive
    description:
      name: bazel_worker
      url: "https://pub.dartlang.org"
    source: hosted
    version: "0.1.25"
```

The full source of this project is available at https://gitlab.com/ChildrenOfUr/coUclient/-/tree/dev

**Comment by jakemac53** [comment]
@johnniwinther can you triage this?

**Comment by Klikini** [comment]
Here's [main.dart](https://gitlab.com/ChildrenOfUr/coUclient/-/blob/cd7fb35d9e57764f96a37a1fa655522259ebfb92/web/main.dart).

It's not just `part 'package:couclient/src/display/windows/go_window.dart';` that it complains about, here's the [full build output](https://gitlab.com/ChildrenOfUr/coUclient/-/jobs/630308536).

**Comment by jakemac53** [comment]
@Klikini shot in the dark but you could try removing this line... https://gitlab.com/ChildrenOfUr/coUclient/-/blob/cd7fb35d9e57764f96a37a1fa655522259ebfb92/web/main.dart#L7.

It shouldn't be confusing things but its possible. It isn't _that_ far off from a language version comment.

**Comment by Klikini** [comment]
@jakemac53 Sure thing: https://gitlab.com/ChildrenOfUr/coUclient/-/commit/4ebb9db451bea392b7ace66f2d3fdc500a6a640d?view=parallel

Unfortunately, it didn't work: https://gitlab.com/ChildrenOfUr/coUclient/-/jobs/634030351

I didn't even think about lines like that causing problems, but when you mentioned it, I could totally see it being an issue lol

**Comment by johnniwinther** [comment]
It is an unusual file layout: The package entry-point is in `web/main.dart` and the part files are in `lib/src/` (normally a package would be fully contained within the `lib` folder). 

What does your `.dart_tool/package_config.json` file look like when you compile the code?

I can repro the problem with `package_config.json` that looks like this:
```
{
  "configVersion": 2,
  "packages": [
    {
      "name": "couclient",
      "rootUri": "../lib/",
      "languageVersion": "2.7"
    }
  ]
}
```
This would tell that the part files are part of package 'couclient' but that `web/main.dart` isn't. Note that the used entry-point is `web/main.dart` and not something like `package:couclient/main.dart`, meaning that the compiler doesn't think `web/main.dart` is part of the package (if it isn't in a subfolder of the `"rootUri"` in the `package_config.json` file).

For a more regular file layout I'd suggest moving the content of `web/main.dart` to `lib/couclient.dart` and then change `web/main.dart` to
```Dart
import 'package:couclient/couclient.dart` as couclient;

main() {
  couclient.main();
}
```

**Comment by jakemac53** [comment]
@johnniwinther Assuming this is a standard pub package (which it appears to be, given it has a pubspec) the rootUri would be the true root of the package, and then it would have a `packageUri` of `lib/`. So that should work, although your are right that the layout is weird.

**Comment by Klikini** [comment]
And we have a successful build! https://gitlab.com/ChildrenOfUr/coUclient/-/jobs/636989081

Thanks, @johnniwinther, I never would've thought about that!

This project has had `main.dart` in `web` instead of `lib` for about 6 years and it only recently broke, so I'm not sure what happened.

---

**To people reading this in the future:** if you see `the language version override has to be the same in the library and its part(s)`, make sure there are no `part` statements in files outside of `lib` that reference files inside it.

I suppose I should have noticed things were weird when we ended up with `part 'package:…';`

**Comment by jakemac53** [comment]
The check you ran into is from the new language versions feature so that is why you haven't seen it before. It looks pretty clearly like a bug in that implementation.

It is probably not super high priority to fix because it is an uncommon edge case, but we should eventually fix it :).

**Comment by johnniwinther** [comment]
I'd like to see the `.dart_tool/package_config.json` file used when you encountered the problem. If it includes `web/main.dart` in the package (for instance by having `"rootUri"` being a parent directory of the `web` directory) then it is a failure of the implementation that we should look into. If the `web` folder _isn't_ included, I do think is working as designed. Whether that is a good design is a different question.

**Comment by Klikini** [comment]
@johnniwinther I followed these steps before copying this file:

1. Checked out the commit before the fix
1. Deleted the entire `.dart_tool` directory
1. Ran `pub get`
1. Ran `webdev build` (which failed, as expected)

```json
{
  "configVersion": 2,
  "packages": [
    {
      "name": "analyzer",
      "rootUri": "file:///C:/Users/andyc/AppData/Roaming/Pub/Cache/hosted/pub.dartlang.org/analyzer-0.35.4",
      "packageUri": "lib/",
      "languageVersion": "2.1"
    },
    {
      "name": "angular",
      "rootUri": "file:///C:/Users/andyc/AppData/Roaming/Pub/Cache/hosted/pub.dartlang.org/angular-5.3.1",
      "packageUri": "lib/",
      "languageVersion": "2.2"
    },
    {
      "name": "angular_ast",
      "rootUri": "file:///C:/Users/andyc/AppData/Roaming/Pub/Cache/hosted/pub.dartlang.org/angular_ast-0.5.9",
      "packageUri": "lib/",
      "languageVersion": "2.2"
    },
    {
      "name": "angular_compiler",
      "rootUri": "file:///C:/Users/andyc/AppData/Roaming/Pub/Cache/hosted/pub.dartlang.org/angular_compiler-0.4.3",
      "packageUri": "lib/",
      "languageVersion": "2.2"
    },
    {
      "name": "angular_forms",
      "rootUri": "file:///C:/Users/andyc/AppData/Roaming/Pub/Cache/hosted/pub.dartlang.org/angular_forms-2.1.2",
      "packageUri": "lib/",
      "languageVersion": "2.2"
    },
    {
      "name": "archive",
      "rootUri": "file:///C:/Users/andyc/AppData/Roaming/Pub/Cache/hosted/pub.dartlang.org/archive-2.0.13",
      "packageUri": "lib/",
      "languageVersion": "2.0"
    },
    {
      "name": "args",
      "rootUri": "file:///C:/Users/andyc/AppData/Roaming/Pub/Cache/hosted/pub.dartlang.org/args-1.6.0",
      "packageUri": "lib/",
      "languageVersion": "2.3"
    },
    {
      "name": "async",
      "rootUri": "file:///C:/Users/andyc/AppData/Roaming/Pub/Cache/hosted/pub.dartlang.org/async-2.4.2",
      "packageUri": "lib/",
      "languageVersion": "2.2"
    },
    {
      "name": "bazel_worker",
      "rootUri": "file:///C:/Users/andyc/AppData/Roaming/Pub/Cache/hosted/pub.dartlang.org/bazel_worker-0.1.25",
      "packageUri": "lib/",
      "languageVersion": "2.3"
    },
    {
      "name": "build",
      "rootUri": "file:///C:/Users/andyc/AppData/Roaming/Pub/Cache/hosted/pub.dartlang.org/build-1.2.2",
      "packageUri": "lib/",
      "languageVersion": "2.3"
    },
    {
      "name": "build_config",
      "rootUri": "file:///C:/Users/andyc/AppData/Roaming/Pub/Cache/hosted/pub.dartlang.org/build_config-0.4.2",
      "packageUri": "lib/",
      "languageVersion": "2.6"
    },
    {
      "name": "build_daemon",
      "rootUri": "file:///C:/Users/andyc/AppData/Roaming/Pub/Cache/hosted/pub.dartlang.org/build_daemon-2.1.4",
      "packageUri": "lib/",
      "languageVersion": "2.6"
    },
    {
      "name": "build_modules",
      "rootUri": "file:///C:/Users/andyc/AppData/Roaming/Pub/Cache/hosted/pub.dartlang.org/build_modules-2.9.0",
      "packageUri": "lib/",
      "languageVersion": "2.8"
    },
    {
      "name": "build_resolvers",
      "rootUri": "file:///C:/Users/andyc/AppData/Roaming/Pub/Cache/hosted/pub.dartlang.org/build_resolvers-1.2.1",
      "packageUri": "lib/",
      "languageVersion": "2.3"
    },
    {
      "name": "build_runner",
      "rootUri": "file:///C:/Users/andyc/AppData/Roaming/Pub/Cache/hosted/pub.dartlang.org/build_runner-1.9.0",
      "packageUri": "lib/",
      "languageVersion": "2.6"
    },
    {
      "name": "build_runner_core",
      "rootUri": "file:///C:/Users/andyc/AppData/Roaming/Pub/Cache/hosted/pub.dartlang.org/build_runner_core-5.1.0",
      "packageUri": "lib/",
      "languageVersion": "2.7"
    },
    {
      "name": "build_web_compilers",
      "rootUri": "file:///C:/Users/andyc/AppData/Roaming/Pub/Cache/hosted/pub.dartlang.org/build_web_compilers-2.10.0",
      "packageUri": "lib/",
      "languageVersion": "2.8"
    },
    {
      "name": "built_collection",
      "rootUri": "file:///C:/Users/andyc/AppData/Roaming/Pub/Cache/hosted/pub.dartlang.org/built_collection-4.3.2",
      "packageUri": "lib/",
      "languageVersion": "2.6"
    },
    {
      "name": "built_value",
      "rootUri": "file:///C:/Users/andyc/AppData/Roaming/Pub/Cache/hosted/pub.dartlang.org/built_value-7.1.0",
      "packageUri": "lib/",
      "languageVersion": "2.3"
    },
    {
      "name": "charcode",
      "rootUri": "file:///C:/Users/andyc/AppData/Roaming/Pub/Cache/hosted/pub.dartlang.org/charcode-1.1.3",
      "packageUri": "lib/",
      "languageVersion": "2.0"
    },
    {
      "name": "checked_yaml",
      "rootUri": "file:///C:/Users/andyc/AppData/Roaming/Pub/Cache/hosted/pub.dartlang.org/checked_yaml-1.0.2",
      "packageUri": "lib/",
      "languageVersion": "2.3"
    },
    {
      "name": "code_builder",
      "rootUri": "file:///C:/Users/andyc/AppData/Roaming/Pub/Cache/hosted/pub.dartlang.org/code_builder-3.4.0",
      "packageUri": "lib/",
      "languageVersion": "2.7"
    },
    {
      "name": "collection",
      "rootUri": "file:///C:/Users/andyc/AppData/Roaming/Pub/Cache/hosted/pub.dartlang.org/collection-1.14.13",
      "packageUri": "lib/",
      "languageVersion": "2.3"
    },
    {
      "name": "convert",
      "rootUri": "file:///C:/Users/andyc/AppData/Roaming/Pub/Cache/hosted/pub.dartlang.org/convert-2.1.1",
      "packageUri": "lib/",
      "languageVersion": "1.17"
    },
    {
      "name": "cou_login",
      "rootUri": "file:///C:/Users/andyc/AppData/Roaming/Pub/Cache/git/cou_login-fa5b88be7ff89b87ffb1bea2b4384a7df6e0b00c/",
      "packageUri": "lib/",
      "languageVersion": "2.0"
    },
    {
      "name": "cou_toolkit",
      "rootUri": "file:///C:/Users/andyc/AppData/Roaming/Pub/Cache/git/cou_toolkit-ce51c87d1b07649702d4d5d6ba24d49c275df02b/",
      "packageUri": "lib/",
      "languageVersion": "2.0"
    },
    {
      "name": "crypto",
      "rootUri": "file:///C:/Users/andyc/AppData/Roaming/Pub/Cache/hosted/pub.dartlang.org/crypto-2.1.5",
      "packageUri": "lib/",
      "languageVersion": "2.3"
    },
    {
      "name": "csslib",
      "rootUri": "file:///C:/Users/andyc/AppData/Roaming/Pub/Cache/hosted/pub.dartlang.org/csslib-0.14.6+1",
      "packageUri": "lib/",
      "languageVersion": "2.0"
    },
    {
      "name": "dart_style",
      "rootUri": "file:///C:/Users/andyc/AppData/Roaming/Pub/Cache/hosted/pub.dartlang.org/dart_style-1.2.4",
      "packageUri": "lib/",
      "languageVersion": "2.0"
    },
    {
      "name": "dnd",
      "rootUri": "file:///C:/Users/andyc/AppData/Roaming/Pub/Cache/hosted/pub.dartlang.org/dnd-1.4.3",
      "packageUri": "lib/",
      "languageVersion": "2.0"
    },
    {
      "name": "firebase",
      "rootUri": "file:///C:/Users/andyc/AppData/Roaming/Pub/Cache/hosted/pub.dartlang.org/firebase-5.0.4",
      "packageUri": "lib/",
      "languageVersion": "2.1"
    },
    {
      "name": "fixnum",
      "rootUri": "file:///C:/Users/andyc/AppData/Roaming/Pub/Cache/hosted/pub.dartlang.org/fixnum-0.10.11",
      "packageUri": "lib/",
      "languageVersion": "2.1"
    },
    {
      "name": "front_end",
      "rootUri": "file:///C:/Users/andyc/AppData/Roaming/Pub/Cache/hosted/pub.dartlang.org/front_end-0.1.14",
      "packageUri": "lib/",
      "languageVersion": "2.1"
    },
    {
      "name": "glob",
      "rootUri": "file:///C:/Users/andyc/AppData/Roaming/Pub/Cache/hosted/pub.dartlang.org/glob-1.2.0",
      "packageUri": "lib/",
      "languageVersion": "2.1"
    },
    {
      "name": "googleapis_auth",
      "rootUri": "file:///C:/Users/andyc/AppData/Roaming/Pub/Cache/hosted/pub.dartlang.org/googleapis_auth-0.2.12",
      "packageUri": "lib/",
      "languageVersion": "2.0"
    },
    {
      "name": "gorgon",
      "rootUri": "file:///C:/Users/andyc/AppData/Roaming/Pub/Cache/git/gorgonDart-0bcec0c14786e96e9cb8660100a6abba34d98ba0/",
      "packageUri": "lib/",
      "languageVersion": "2.0"
    },
    {
      "name": "graphs",
      "rootUri": "file:///C:/Users/andyc/AppData/Roaming/Pub/Cache/hosted/pub.dartlang.org/graphs-0.2.0",
      "packageUri": "lib/",
      "languageVersion": "2.0"
    },
    {
      "name": "http",
      "rootUri": "file:///C:/Users/andyc/AppData/Roaming/Pub/Cache/hosted/pub.dartlang.org/http-0.11.3+17",
      "packageUri": "lib/",
      "languageVersion": "2.0"
    },
    {
      "name": "http_multi_server",
      "rootUri": "file:///C:/Users/andyc/AppData/Roaming/Pub/Cache/hosted/pub.dartlang.org/http_multi_server-2.2.0",
      "packageUri": "lib/",
      "languageVersion": "2.1"
    },
    {
      "name": "http_parser",
      "rootUri": "file:///C:/Users/andyc/AppData/Roaming/Pub/Cache/hosted/pub.dartlang.org/http_parser-3.1.4",
      "packageUri": "lib/",
      "languageVersion": "2.3"
    },
    {
      "name": "intl",
      "rootUri": "file:///C:/Users/andyc/AppData/Roaming/Pub/Cache/hosted/pub.dartlang.org/intl-0.15.8",
      "packageUri": "lib/",
      "languageVersion": "2.0"
    },
    {
      "name": "io",
      "rootUri": "file:///C:/Users/andyc/AppData/Roaming/Pub/Cache/hosted/pub.dartlang.org/io-0.3.4",
      "packageUri": "lib/",
      "languageVersion": "2.0"
    },
    {
      "name": "js",
      "rootUri": "file:///C:/Users/andyc/AppData/Roaming/Pub/Cache/hosted/pub.dartlang.org/js-0.6.2",
      "packageUri": "lib/",
      "languageVersion": "2.0"
    },
    {
      "name": "json_annotation",
      "rootUri": "file:///C:/Users/andyc/AppData/Roaming/Pub/Cache/hosted/pub.dartlang.org/json_annotation-2.4.0",
      "packageUri": "lib/",
      "languageVersion": "2.2"
    },
    {
      "name": "json_serializable",
      "rootUri": "file:///C:/Users/andyc/AppData/Roaming/Pub/Cache/hosted/pub.dartlang.org/json_serializable-3.1.0",
      "packageUri": "lib/",
      "languageVersion": "2.2"
    },
    {
      "name": "kernel",
      "rootUri": "file:///C:/Users/andyc/AppData/Roaming/Pub/Cache/hosted/pub.dartlang.org/kernel-0.3.14",
      "packageUri": "lib/",
      "languageVersion": "2.0"
    },
    {
      "name": "libld",
      "rootUri": "file:///C:/Users/andyc/AppData/Roaming/Pub/Cache/git/libld-5f75df933097ff328bd040d98ad8bb5236e94e94/",
      "packageUri": "lib/",
      "languageVersion": "2.0"
    },
    {
      "name": "logging",
      "rootUri": "file:///C:/Users/andyc/AppData/Roaming/Pub/Cache/hosted/pub.dartlang.org/logging-0.11.4",
      "packageUri": "lib/",
      "languageVersion": "2.0"
    },
    {
      "name": "matcher",
      "rootUri": "file:///C:/Users/andyc/AppData/Roaming/Pub/Cache/hosted/pub.dartlang.org/matcher-0.12.8",
      "packageUri": "lib/",
      "languageVersion": "2.4"
    },
    {
      "name": "meta",
      "rootUri": "file:///C:/Users/andyc/AppData/Roaming/Pub/Cache/hosted/pub.dartlang.org/meta-1.2.2",
      "packageUri": "lib/",
      "languageVersion": "1.12"
    },
    {
      "name": "mime",
      "rootUri": "file:///C:/Users/andyc/AppData/Roaming/Pub/Cache/hosted/pub.dartlang.org/mime-0.9.6+3",
      "packageUri": "lib/",
      "languageVersion": "2.0"
    },
    {
      "name": "node_interop",
      "rootUri": "file:///C:/Users/andyc/AppData/Roaming/Pub/Cache/hosted/pub.dartlang.org/node_interop-1.1.1",
      "packageUri": "lib/",
      "languageVersion": "2.0"
    },
    {
      "name": "node_io",
      "rootUri": "file:///C:/Users/andyc/AppData/Roaming/Pub/Cache/hosted/pub.dartlang.org/node_io-1.1.1",
      "packageUri": "lib/",
      "languageVersion": "2.2"
    },
    {
      "name": "package_config",
      "rootUri": "file:///C:/Users/andyc/AppData/Roaming/Pub/Cache/hosted/pub.dartlang.org/package_config-1.9.3",
      "packageUri": "lib/",
      "languageVersion": "2.7"
    },
    {
      "name": "package_resolver",
      "rootUri": "file:///C:/Users/andyc/AppData/Roaming/Pub/Cache/hosted/pub.dartlang.org/package_resolver-1.0.10",
      "packageUri": "lib/",
      "languageVersion": "2.1"
    },
    {
      "name": "path",
      "rootUri": "file:///C:/Users/andyc/AppData/Roaming/Pub/Cache/hosted/pub.dartlang.org/path-1.7.0",
      "packageUri": "lib/",
      "languageVersion": "2.0"
    },
    {
      "name": "pedantic",
      "rootUri": "file:///C:/Users/andyc/AppData/Roaming/Pub/Cache/hosted/pub.dartlang.org/pedantic-1.9.2",
      "packageUri": "lib/",
      "languageVersion": "2.1"
    },
    {
      "name": "petitparser",
      "rootUri": "file:///C:/Users/andyc/AppData/Roaming/Pub/Cache/hosted/pub.dartlang.org/petitparser-3.0.4",
      "packageUri": "lib/",
      "languageVersion": "2.7"
    },
    {
      "name": "pool",
      "rootUri": "file:///C:/Users/andyc/AppData/Roaming/Pub/Cache/hosted/pub.dartlang.org/pool-1.4.0",
      "packageUri": "lib/",
      "languageVersion": "2.0"
    },
    {
      "name": "protobuf",
      "rootUri": "file:///C:/Users/andyc/AppData/Roaming/Pub/Cache/hosted/pub.dartlang.org/protobuf-1.0.1",
      "packageUri": "lib/",
      "languageVersion": "2.0"
    },
    {
      "name": "pub_semver",
      "rootUri": "file:///C:/Users/andyc/AppData/Roaming/Pub/Cache/hosted/pub.dartlang.org/pub_semver-1.4.4",
      "packageUri": "lib/",
      "languageVersion": "2.0"
    },
    {
      "name": "pubspec_parse",
      "rootUri": "file:///C:/Users/andyc/AppData/Roaming/Pub/Cache/hosted/pub.dartlang.org/pubspec_parse-0.1.5",
      "packageUri": "lib/",
      "languageVersion": "2.2"
    },
    {
      "name": "quiver",
      "rootUri": "file:///C:/Users/andyc/AppData/Roaming/Pub/Cache/hosted/pub.dartlang.org/quiver-2.1.3",
      "packageUri": "lib/",
      "languageVersion": "2.0"
    },
    {
      "name": "scproxy",
      "rootUri": "file:///C:/Users/andyc/AppData/Roaming/Pub/Cache/git/scproxy-6b8c44b855595df33327884ef503251f3ed2f705/",
      "packageUri": "lib/",
      "languageVersion": "2.0"
    },
    {
      "name": "scratch_space",
      "rootUri": "file:///C:/Users/andyc/AppData/Roaming/Pub/Cache/hosted/pub.dartlang.org/scratch_space-0.0.4+2",
      "packageUri": "lib/",
      "languageVersion": "2.3"
    },
    {
      "name": "service_worker",
      "rootUri": "file:///C:/Users/andyc/AppData/Roaming/Pub/Cache/hosted/pub.dartlang.org/service_worker-0.2.4",
      "packageUri": "lib/",
      "languageVersion": "2.0"
    },
    {
      "name": "shelf",
      "rootUri": "file:///C:/Users/andyc/AppData/Roaming/Pub/Cache/hosted/pub.dartlang.org/shelf-0.7.7",
      "packageUri": "lib/",
      "languageVersion": "2.1"
    },
    {
      "name": "shelf_web_socket",
      "rootUri": "file:///C:/Users/andyc/AppData/Roaming/Pub/Cache/hosted/pub.dartlang.org/shelf_web_socket-0.2.3",
      "packageUri": "lib/",
      "languageVersion": "2.0"
    },
    {
      "name": "slack",
      "rootUri": "file:///C:/Users/andyc/AppData/Roaming/Pub/Cache/git/dart-slack-cf74d4dd13caa0f1e092a46a9912de9913e615f2/",
      "packageUri": "lib/",
      "languageVersion": "2.0"
    },
    {
      "name": "source_gen",
      "rootUri": "file:///C:/Users/andyc/AppData/Roaming/Pub/Cache/hosted/pub.dartlang.org/source_gen-0.9.4+4",
      "packageUri": "lib/",
      "languageVersion": "2.1"
    },
    {
      "name": "source_maps",
      "rootUri": "file:///C:/Users/andyc/AppData/Roaming/Pub/Cache/hosted/pub.dartlang.org/source_maps-0.10.9",
      "packageUri": "lib/",
      "languageVersion": "2.0"
    },
    {
      "name": "source_span",
      "rootUri": "file:///C:/Users/andyc/AppData/Roaming/Pub/Cache/hosted/pub.dartlang.org/source_span-1.7.0",
      "packageUri": "lib/",
      "languageVersion": "2.6"
    },
    {
      "name": "stack_trace",
      "rootUri": "file:///C:/Users/andyc/AppData/Roaming/Pub/Cache/hosted/pub.dartlang.org/stack_trace-1.9.5",
      "packageUri": "lib/",
      "languageVersion": "2.0"
    },
    {
      "name": "stagexl",
      "rootUri": "file:///C:/Users/andyc/AppData/Roaming/Pub/Cache/hosted/pub.dartlang.org/stagexl-1.4.5",
      "packageUri": "lib/",
      "languageVersion": "2.7"
    },
    {
      "name": "stream_channel",
      "rootUri": "file:///C:/Users/andyc/AppData/Roaming/Pub/Cache/hosted/pub.dartlang.org/stream_channel-2.0.0",
      "packageUri": "lib/",
      "languageVersion": "2.0"
    },
    {
      "name": "stream_transform",
      "rootUri": "file:///C:/Users/andyc/AppData/Roaming/Pub/Cache/hosted/pub.dartlang.org/stream_transform-1.2.0",
      "packageUri": "lib/",
      "languageVersion": "2.6"
    },
    {
      "name": "string_scanner",
      "rootUri": "file:///C:/Users/andyc/AppData/Roaming/Pub/Cache/hosted/pub.dartlang.org/string_scanner-1.0.5",
      "packageUri": "lib/",
      "languageVersion": "2.0"
    },
    {
      "name": "term_glyph",
      "rootUri": "file:///C:/Users/andyc/AppData/Roaming/Pub/Cache/hosted/pub.dartlang.org/term_glyph-1.1.0",
      "packageUri": "lib/",
      "languageVersion": "1.8"
    },
    {
      "name": "timing",
      "rootUri": "file:///C:/Users/andyc/AppData/Roaming/Pub/Cache/hosted/pub.dartlang.org/timing-0.1.1+2",
      "packageUri": "lib/",
      "languageVersion": "2.2"
    },
    {
      "name": "transmit",
      "rootUri": "file:///C:/Users/andyc/AppData/Roaming/Pub/Cache/git/transmit-5b8e99e9bd80273de4064040d1836decd33fe005/",
      "packageUri": "lib/",
      "languageVersion": "2.0"
    },
    {
      "name": "typed_data",
      "rootUri": "file:///C:/Users/andyc/AppData/Roaming/Pub/Cache/hosted/pub.dartlang.org/typed_data-1.2.0",
      "packageUri": "lib/",
      "languageVersion": "2.4"
    },
    {
      "name": "watcher",
      "rootUri": "file:///C:/Users/andyc/AppData/Roaming/Pub/Cache/hosted/pub.dartlang.org/watcher-0.9.7+15",
      "packageUri": "lib/",
      "languageVersion": "2.2"
    },
    {
      "name": "web_socket_channel",
      "rootUri": "file:///C:/Users/andyc/AppData/Roaming/Pub/Cache/hosted/pub.dartlang.org/web_socket_channel-1.1.0",
      "packageUri": "lib/",
      "languageVersion": "2.0"
    },
    {
      "name": "xml",
      "rootUri": "file:///C:/Users/andyc/AppData/Roaming/Pub/Cache/hosted/pub.dartlang.org/xml-4.2.0",
      "packageUri": "lib/",
      "languageVersion": "2.7"
    },
    {
      "name": "yaml",
      "rootUri": "file:///C:/Users/andyc/AppData/Roaming/Pub/Cache/hosted/pub.dartlang.org/yaml-2.2.1",
      "packageUri": "lib/",
      "languageVersion": "2.4"
    },
    {
      "name": "couclient",
      "rootUri": "../",
      "packageUri": "lib/",
      "languageVersion": "2.3"
    }
  ],
  "generated": "2020-07-15T13:30:18.392093Z",
  "generator": "pub",
  "generatorVersion": "2.8.4"
}
```

**Comment by johnniwinther** [comment]
@Klikini thanks for the detailed info. It looks like your `.dart_tool/package_config.json` file should have included the `web/main.js` in the package. I'll look into why that failed.

**Comment by johnniwinther** [comment]
I can reproduce the failure in `2.8.4`. The issue has already been fixed by https://dart-review.googlesource.com/c/sdk/+/152151 which included in `2.9.0-20.0.dev`. I'm adding a regression test for it in https://dart-review.googlesource.com/c/sdk/+/154680
